### PR TITLE
Like filter normalize

### PR DESCRIPTION
**Pull request by janiukjf**
Added an optional third parameter to the like filter to allow you to pass in a regular expression for normalization of the search and source strings if needed.

### CODE REVIEW

File: lib/data_filter/filter_set.rb [hunk 1]
```diff
@@ -92,9 +92,9 @@ def create(&block)
       #   to filter
       # @option opts [Object] :by the value we want to use when
       #   filtering the data item
-      def like_filter(field_sym, opts = {})
+      def like_filter(field_sym, opts = {}, normalize_regex = /[^\w\s]/)
```
Comment (jbodah) [inline lib/data_filter/filter_set.rb]: What about having this be another option instead? That way it'd look something like:

```rb
like_filter by: :name, normalize_regex: /[^\w\s^.^@^-]/
```

Also could you add to the doc block above as well as to the examples in `README.md`?

Comment (janiukjf) [inline lib/data_filter/filter_set.rb]: I'm totally ok with it being another option. I'll update that, the doc block and the examples.

**Comment by jbodah** [comment]
Looks good! will push this up to rubygems in an hour or so

**Comment by thebmo** [comment]
👍

**Comment by jbodah** [comment]
Pushed in v0.4.0

**Pull request merged**